src/apiServices/AnalysesService.ts

```diff
@@ -4,34 +4,32 @@ import { errorHandler } from '../util/errorHandler'
 import { getHeaders } from '../util/getHeaders'
 import { submitBytecodeRequest } from '../util/submitBytecodeRequest'
 
-import { getTokensNode } from '../node'
-
 import { API_URL_PRODUCTION, API_URL_STAGING, tokenLocation } from '../util/constants'
 
-import { JwtTokensInterface, SubmitContractRes } from '..'
+import { isTokenValid } from '../util/isTokenValid'
+
+import { SubmitContractRes } from '..'
 
 import { submitContractRequest } from '../analyses/submitContractRequest'
 
 export class AnalysesService {
     public token
     private apiUrl: string = API_URL_PRODUCTION
 
-    constructor(token?) {
-        this.token = token
+    constructor(token) {
+        if (isTokenValid(token)) {
+            this.token = token
+        } else {
+            throw new Error('Access token has expired or is invalid!')
+        }
     }
 
     public async getAnalysesList() {
         try {
-            let headers;
-            if (this.token) {
-                headers = getHeaders(this.token)
-            } else {
-                const jwtTokens: JwtTokensInterface = getTokensNode(tokenLocation)
-                headers = getHeaders(jwtTokens.access)
-            }
+            const headers = getHeaders(this.token)
```

Comment:
Since the headers are always the same, why don't we set `this.headers = getHeaders(this.token)` in the constructor?

Reply:
We can also write a wrapper for the `getRequest` and `postRequest` methods that read the this.token or this.headers and send the request with the correct headers, as we do reuse that logic in multiple places.